### PR TITLE
Deal with variably padded addresses

### PR DESCRIPTION
**Pull request by kennsippell**
https://app.clubhouse.io/augur/story/5104/deal-with-variably-padded-addresses

### CODE REVIEW

File: src/server/address-format-reviver.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/*
+A reviver is a function that prescribes how the value originally produced by JSON.parse is transformed, before being returned.
+*/
+
+import { formatEthereumAddress } from "speedomatic";
+
+export const inputsExpectedAsAddress: Array<string> = [
```
Comment (pgebheim) [inline src/server/address-format-reviver.ts]: Use the Object-as-set trick to turn this into O(1) Lookups.

File: src/server/address-format-reviver.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  [key: string]: boolean;
+}
+
+export const inputsExpectedAsAddress: StringToBoolMap = {
+  account: true,
+  category: true,
+  creator: true,
+  designatedReporter: true,
+  marketID: true,
+  marketIDs: true,
```
Comment (epheph) [inline src/server/address-format-reviver.ts]: I was going to comment that this won't work, but after reading formatEthereumAddress and seeing it works with arrays, looks good.

File: src/server/address-format-reviver.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  reporter: true,
+  reportingWindow: true,
+  result: true,
```
Comment (epheph) [inline src/server/address-format-reviver.ts]: result? I don't know if this is necessary as a parameter to a getter

File: src/server/address-format-reviver.ts (hunk 2 above)
Comment (kennsippell) [reply src/server/address-format-reviver.ts]: ya. got a unit test for that case also

File: src/server/run-websocket-server.ts [hunk 4]
```diff
@@ -36,7 +37,7 @@ export function runWebsocketServer(db: Knex, webSocketConfigs: WebSocketConfigs)
       websocket.on("message", (data: WebSocket.Data): void => {
         let message: any;
         try {
-          message = JSON.parse(data as string);
+          message = JSON.parse(data as string, addressFormatReviver);
```
Comment (epheph) [inline src/server/run-websocket-server.ts]: nice dealing with it in the parser

File: src/server/address-format-reviver.ts (hunk 3 above)
Comment (kennsippell) [reply src/server/address-format-reviver.ts]: ic -- it is the name of the variable in the "callback" from the definitions. removing

**Pull request merged**